esp/smc/SMCLib/TpWrapper.cpp

```diff
@@ -29,6 +29,17 @@
 #include "dautils.hpp"
 #include "dameta.hpp"
 
+static unsigned reloadConifgCBId = (unsigned)-1;
```

Comment:
If this is going to happen frequently, I wonder if we should have a helper class that can be used to
a) remove the hook on destruction
b) (conditionally) create the hook from a provided function.

So then would become
```
static ConfigUpdateHook reloadConfigHook;
reloadConfigHook.install(func);

```
or
`reloadConfigHook.installOnce(func);
`?

Could also have a constructor that took a function which called the hook, but couldn't be used everywhere because it is too early if called statically.

Reply:
Yes, I think worth it, I've implemented it (last commit in https://github.com/hpcc-systems/HPCC-Platform/pull/14966),
but think it should probably be in a separate PR after this one, as it's general, separate from this eclccserver change and modified the existing hooks

Reply:
It should be a separate PR so that you can clean up the other existing hooks.